backend/layers/processing/schema_migration.py

```diff
@@ -60,20 +60,22 @@ def gather_collections(self, auto_publish: bool) -> List[Dict[str, str]]:
         """
         response = []
 
-        has_revision = set()
+        has_migration_revision = set()
         # iterates over unpublished collections first, so published versions are skipped if there is an active revision
         for collection in self.fetch_collections():
             _resp = {}
-            if collection.is_published() and collection.collection_id.id in has_revision:
+            if collection.is_published() and collection.collection_id.id in has_migration_revision:
                 continue
 
             if collection.is_published():
                 # published collection without an active revision
                 _resp["can_publish"] = str(True)
             elif collection.is_unpublished_version():
                 # active revision of a published collection.
-                has_revision.add(collection.collection_id.id)  # revision found, skip published version
                 _resp["can_publish"] = str(False)
+                if collection.is_migration_revision:
```

Comment:
I don't think this flag would have been set yet since it isn't add until `collection_migrate`. Oh is this incase we rerun migration a second time? Add a comment to reflect the specific cases this is intended to cover.

Reply:
Yup, thats the case we want to cover with this. I can add a comment